src/Microsoft.SqlTools.ServiceLayer/Connection/ReliableConnection/CachedServerInfo.cs

```diff
@@ -115,6 +116,80 @@ public static void AddOrUpdateIsAzure(string  dataSource, bool isAzure)
             }
         }
 
+        public static void AddOrUpdateIsSqlDw(IDbConnection connection, bool isSqlDw)
+        {
+            Validate.IsNotNull(nameof(connection), connection);
+
+            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(connection.ConnectionString);
+            AddOrUpdateIsSqlDw(builder.DataSource, isSqlDw);
+        }
+
+        public static void AddOrUpdateIsSqlDw(string dataSource, bool isSqlDw)
```

Comment:
>AddOrUpdateIsSqlDw [](start = 27, length = 18)

this looks basically the same as AddOrUpdateIsAzure..is it possible to factor common logic into a shared method?